test/doc/websocket_2_handshaking.cpp

```diff
@@ -132,6 +132,74 @@ snippets()
 
     //]
     }
+
+    {
+        //[code_websocket_2_6
+
+        // This buffer is required for reading HTTP messages
+        flat_buffer buffer;
+
+        // Read the HTTP request ourselves
+        http::request<http::string_body> req;
+        http::read(sock, buffer, req);
+
+        // See if it's a WebSocket upgrade request
+        if(websocket::is_upgrade(req))
+        {
+            // tokenize the Sec-Websocket-Protocol header offered by the client
+            http::token_list offered { req[http::field::sec_websocket_protocol] };
+
+            // an array of protocols supported by this server
+            // in descending order of preference
+            static const std::array<string_view, 3>
+                supported = {
+                    "v3.my.chat",
+                    "v2.my.chat",
+                    "v1.my.chat"
+                };
+
+            for (auto proto : supported)
```

Comment:
I would factor out the subprotocol checks into its own function with this signature:
```
string_view
select_subprotocol( string_view subprotocols );
```

Reply:
wow!